### PR TITLE
devguide: add page on Frame support - v10

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata/pull/6982

Describe changes:
- add note explaining when to create pdu frames
- fix wording issue in a paragraph

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4697

Compiled version can be checked at
https://stream-frame-support-v10.surge.sh/extending/app-layer/app-layer-frames.html

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on tlpw2_single_alerts_cmp.

ERROR: QA failed on tlpw2_autofp_alerts_cmp.

ERROR: QA failed on tlpr1_alerts_cmp.

ERROR: QA failed on ips_afp_drop_chk.
  

Pipeline 6247

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+Baseline
+========
+
+- `Suricata rules format <https://suricata.readthedocs.io/en/latest/rules/intro.html>`_
+- It helps to understand the basics of the `Multi-pattern rule matching <https://suricata.readthedocs.io/en/latest/configuration/suricata-yaml.html?highlight=multi%20pattern%20matcher#detection-engine>`_ algorithm, used for rule filtering and to address engine performance
+
+General Concepts
+================
+
+Frame support was introduced with Suricata-7.0. Up until 6.0x, Suricata's architecture and state of parsers meant that the network traffic available to the detection engine was just a stream of data, without detail about higher level parsers.
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:16]: nit: Suricata 7.0

Comment (victorjulien) [reply doc/devguide/extending/app-layer/app-layer-frames.rst:16]: nit: 6.0.x (extra dot)

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+Application Layer Frame Support
+*******************************
+
+.. contents:: Table of Contents
+
+Baseline
+========
+
+- `Suricata rules format <https://suricata.readthedocs.io/en/latest/rules/intro.html>`_
+- It helps to understand the basics of the `Multi-pattern rule matching <https://suricata.readthedocs.io/en/latest/configuration/suricata-yaml.html?highlight=multi%20pattern%20matcher#detection-engine>`_ algorithm, used for rule filtering and to address engine performance
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:11]: Not really sure ppl really need to understand MPM. Doesn't seem they need to read this first to understand frames.

**Review by victorjulien — CHANGES_REQUESTED** [review]
Minor comments. I think we're almost ready.

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    :caption: rust/src/smb/smb.rs
+    :language: rust
+    :start-after: // app-layer-frame-documentation tag start: parse_tcp_data_ts
+    :end-before: // app-layer-frame-documentation tag end: parse_tcp_data_ts
+    :lines: 2-5, 59-
+    :dedent: 4
+
+.. note:: when to create PDU frames
+
+    The standard approach we follow for frame registration is that a frame ``pdu`` will always be created, regardless of parser status (in practice, before the parser is called). The other frames are then created when and if only the parser succeeds.
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/devguide/extending/app-layer/app-layer-frames.rst:96]: What if the examples were changed to show snippets from the SIP parser?

**Comment by jufajardini** [comment]
followed by: https://github.com/OISF/suricata/pull/7007

**Pull request closed without merging**